[#90] feature: 탑색 탭 추가

wongbingg opened the pull request:
## 📌 Summary
<!-- PR 요약을 써주세요. -->
탐색 탭 추가 작업 


## ✍️ Description
<!-- PR에 대한 자세한 설명을 써주세요. -->
- close: #90 

## 💡 PR Point
<!-- 코드를 작성할 때 고민했던 부분을 적어주세요 -->
- Home 하위에 Recommend 와 Explore 를 정의하여 두 탭을 관리하도록 구현했습니다. 
- BottomSheet 와 같이 전체화면 dimming 배경이 필요한 요소들은 Home에서 띄워주도록 설정했습니다.
- 탐색 탭 ) 아래 작업 미완료 상태입니다
    - [x] 셀 탭시 보여줄 애니메이션, 화면 구현
    - [x] 각 셀 배경색 지정
    - [ ] 페이징 처리


## 📚 Reference 
<!-- 참고할 만한 자료가 있다면 링크나 시각 자료를 달아주세요. -->

## 🔥 Test
<!-- Test -->
|Recommend|Explore|ExploreCardModalView|
|:---:|:---:|:---:|
|<img  width="200" src="https://github.com/user-attachments/assets/284d2a68-2176-4a21-a5eb-fb512eef30b3">|<img width="200" src="https://github.com/user-attachments/assets/6cd2f6c2-3315-4f21-9648-58e5edab61b1">|<img width="200" src="https://github.com/user-attachments/assets/485c13b1-1683-4091-857c-b3905753ddbc">|

wongbingg committed:
[#90] feat: Explore 탭 카드 조회 API 추가

- Explore 탭에서 다루는 카드는 ExploreCard 로 네이밍

wongbingg committed:
[#90] refactor: 기존 메인화면을 Recommend 화면으로 새롭게 생성

- Reducer에 Delegate 추가
  - 상위뷰인 Home 에 이벤트를 전달할 때 사용

wongbingg committed:
[#90] feat: Explore 화면 및 Reducer 구현

wongbingg committed:
[#90] feat: Home 재구성 및 커스텀 네비게이션바 추가

- recommend(추천탭) 과 explore(탐색탭) 을 관리하는 컨테이너 Reducer 역할 부여

wongbingg committed:
[#90] refactor: AppView 재구성

- RemoteConfigManager 파일분리 및 DependencyKey로 의존성 주입

LeeJoEun-01 commented on NewsLetter/NewsLetter/Source/Application/Feature/Home/Component/CustomNativationBar.swift:
```diff
@@ -0,0 +1,58 @@
+
+import SwiftUI
+
+struct CustomNativationBar: View {
```
CustomNativationBar → CustomNavigationBar 오탈자로 수정이 필요해 보입니다!

wongbingg replied:
수정 완료했습니다

LeeJoEun-01 commented on NewsLetter/NewsLetter/Source/Application/AppReducer.swift:
```diff
@@ -9,18 +9,46 @@ import ComposableArchitecture
+            case .remoteConfigResponse(.failure(let error)):
+                state.isLoading = false
+                print("Remote Config Fetch Error: \(error.localizedDescription)")
+                return .none
```
```swift
    let defaultFlags: Flags = (colorFlag: "B", mainDescFlag: "F")
    return .send(.home(.setFlags(defaultFlags)))
```

서버에서 원격 설정을 못 받아와도
항상 HomeReducer에는 "B", "F" 같은 안전한 기본 플래그가 내려가도록 수정이 필요할 것 같습니다.

wongbingg replied:
제시해주신 코드를 그대로 활용하여 수정했습니다 👍

LeeJoEun-01 commented on NewsLetter/NewsLetter/Source/Application/AppReducer.swift:
```diff
@@ -9,18 +9,46 @@ import ComposableArchitecture
+                    await send(.remoteConfigResponse(
+                        Result { try await self.remoteConfigClient.fetch() }
+                    ))
+                }
```
remoteConfigResponse가 오기 전까지는 isLoading이 계속 `true`라서, 첫 진입 후 응답이 오기 전에 `.onAppear`가 여러 번 호출되면 remoteConfigClient.fetch()가 그만큼 중복으로 실행될 수 있습니다.

요청 “시작 여부”와 “로딩 상태”를 분리해서,
hasRequestedRemoteConfig 같은 별도 플래그를 두고 한 번만 요청을 보내거나
혹은 이펙트를 시작하는 시점에 isLoading을 먼저 false로 내려서 이후 .onAppear에서는 더 이상 호출되지 않도록
한 번 더 방어해 주면 레이스를 줄이고 불필요한 원격 설정 호출을 막을 수 있을 것 같아요 !!

wongbingg replied:
별도 플래그를 추가하여 중복 호출이 되지 않도록 설정했습니다

LeeJoEun-01 reviewed (commented):
새로 추가할 것도 많고 수정할 것도 많으셨을텐데 고생하셨습니다 👏🏻👏🏻

wongbingg committed:
[#90] chore: 오탈자 수정

wongbingg committed:
[#90] fix: onAppear 중복 호출 방지용 방어코드 작성

wongbingg committed:
[#90] refactor: ExploreCard 모델 수정

- summary 데이터까지 받아오도록 수정
- Card 타입 변환 메서드 추가

wongbingg committed:
[#90] refactor: CarouselCard 모델 수정

- 사용처에 따른 분기처리를 위해 플래그 변수 isShareEnable 추가 (추천탭: 공유버튼O, 탐색탭: 공유버튼X)

wongbingg committed:
[#90] feat: ExploreCardModalView 생성

- 탐색 탭에서 셀 터치 시, 보여줄 화면
- 셀 배경색 파라미터 추가

wongbingg committed:
[#90] refactor: ExploreView 셀 탭 액션 추가

- delegate 를 통해 상위 뷰인 HomeView로 ModalView present 책임 위임
- 페이지네이션 백엔드 논의 필요 (사이즈 초과시, 500 에러)

wongbingg committed:
[#90] feat: HomView에 ExploreCardModalView overlay 추가

wongbingg committed:
[#90] feat: ExploreCard API 하루단위 캐싱 적용

- 실시간 업데이트되가 필요한 정보가 아니므로 캐싱처리

wongbingg committed:
[#90] fix: ExploreCard 언어정보 숨김처리

- 현재는 language정보가 없어 "기타" 로 표출되는 것이 어색하여 숨김처리
